### PR TITLE
docs: document boundary events migration limitation

### LINKED ISSUE #3089: Update Process Instance Migration docs for event subscription rejections
**Description**

There are newly introduced changes that needs to be updated in our MigrateProcessInstance gRPC docs:

1) Endpoint rejection cases needs to be updated with the changes that was introduced with https://github.com/camunda/zeebe/issues/15405 and https://github.com/camunda/zeebe/issues/15406.
2) Multi tenancy behaviour needs to be mentioned under the endpoint name. The following needs to mentioned for the `NOT_FOUND` case: `No process instance exists with the given key for the tenants the user is authorized to work with.`

### PR DESCRIPTION
**Pull request by abbasadel**
## Description
List PR document the limitation that resulted by [Revert Migrate task instances with recreated event subscriptions](https://github.com/camunda/zeebe/issues/15659)

Closes #3089 

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/concepts/process-instance-migration.md [hunk 1]
```diff
@@ -177,6 +177,7 @@ The following limitations exist that may be supported in future versions:
 - The following scenarios cannot be migrated:
   - A process instance with an incident
   - A process instance that is started from a call activity, i.e. a child process instance
+  - A process instance with a service task when a boundary event is active
```
Comment (saig0) [inline docs/components/concepts/process-instance-migration.md:180]: :x: The limitation is not completely correct.

```suggestion
  - A process instance with an active service task that has a boundary event
```

**Review by saig0 — CHANGES_REQUESTED** [review]
@abbasadel thank you for adding the limitation. :+1: 

I added a suggestion to clarify the limitation.

File: docs/components/concepts/process-instance-migration.md (hunk 1 above)
Comment (berkaycanbc) [reply docs/components/concepts/process-instance-migration.md:180]: @abbasadel Following limitations should also be added:

```md
- A target process definition with a service task that has a boundary event
- A process instance that contains event subprocess
- A target process definition that contains event subprocess
```

**Comment by berkaycanbc** [comment]
@abbasadel I added closes tag for the related issue.

**Comment by berkaycanbc** [comment]
@abbasadel Following updates for MigrateProcessInstance gRPC needs to be added to this PR as well:

Under [GRPC_STATUS_NOT_FOUND](https://docs.camunda.io/docs/next/apis-tools/grpc/#grpc_status_not_found-6):
- No process instance exists with the given key for the tenants the user is authorized to work with.

Cc: @saig0

**Comment by abbasadel** [comment]
Thanks guys for your input. Please have a look now

**Review by saig0 — APPROVED** [review]
@abbasadel  LGTM :+1:

**Review by berkaycanbc — APPROVED** [review]
Nice work @abbasadel, thanks! 🚀

**Comment by abbasadel** [comment]
bors merge

**Comment by abbasadel** [comment]
@christinaausley please go ahead and merge this whenever you like

**Pull request merged**